File: src/main/java/com/beat/domain/booking/application/dto/TicketRefundRequest.java
package com.beat.domain.booking.application.dto;

import java.util.List;

public record TicketRefundRequest(
	Long performanceId,
	List<Booking> bookingList
) {
	public static TicketRefundRequest of(Long performanceId, List<Booking> bookingList) {
		return new TicketRefundRequest(performanceId, bookingList);
	}

	public static record Booking(
		long bookingId
	) {
	}
}

File: apis/src/main/kotlin/com/beat/apis/ticket/api/request/TicketRefundRequest.kt
package com.beat.apis.ticket.api.request

import jakarta.validation.Valid
import jakarta.validation.constraints.NotNull

data class TicketRefundRequest(
    @field:NotNull val performanceId: Long?,
    @field:NotNull @field:Valid val bookingList: List<@Valid Booking>?,
) {
    data class Booking(
        @field:NotNull val bookingId: Long?,
    )
}
